app/src/main/java/ch/epfl/sdp/appart/user/UserViewModel.java

```diff
@@ -53,8 +52,42 @@ public void putUser(User user) {
      * @param user the user to update in database
      */
     public void updateUser(User user) {
-        CompletableFuture<Boolean> updateUser = db.updateUser(user, mUri.getValue());
-        updateUser.thenAccept(mUpdateCardConfirmed::setValue);
+        CompletableFuture<Boolean> updateUser = db.updateUser(user);
+        updateUser.thenAccept(mUpdateUserConfirmed::setValue);
+    }
+
+    /**
+     * Update the user image the database and updates the LiveData
+     * the uri for the image is stored in profileImageUri attribute above
+     *
+     * @param userId the id of the user
+     */
+    public void updateImage(String userId){
+        StringBuilder imageName = new StringBuilder();
+        imageName
+                .append(FirebaseLayout.PROFILE_IMAGE_NAME)
+                .append(System.currentTimeMillis())
+                .append(FirebaseLayout.JPEG);
+
+        StringBuilder imagePath = new StringBuilder();
+        imagePath
+                .append(FirebaseLayout.USERS_DIRECTORY)
+                .append(FirebaseLayout.SEPARATOR)
+                .append(userId);
+
+        CompletableFuture<Boolean> updateImage = db.putImage(profileImageUri, imageName.toString(), imagePath.toString());
```

Comment:
Aren't path and name inverted ?

Reply:
no, the signature and arguments position is correct I think: `FirestoreImageHelper.putImage(uri, name, path)`

Reply:
Yes I checked yesterday and its true sorry.